source/fe/fe_series_fourier.cc

```diff
@@ -65,14 +65,16 @@ namespace
   integrate(const FiniteElement<dim, spacedim> &fe,
             const Quadrature<dim> &             quadrature,
             const Tensor<1, dim> &              k_vector,
-            const unsigned int                  j)
+            const unsigned int                  j,
+            const unsigned int                  component)
   {
     std::complex<double> sum = 0;
     for (unsigned int q = 0; q < quadrature.size(); ++q)
       {
         const Point<dim> &x_q = quadrature.point(q);
         sum += std::exp(std::complex<double>(0, 1) * (k_vector * x_q)) *
-               fe.shape_value(j, x_q) * quadrature.weight(q);
+               fe.shape_value_component(j, x_q, component) *
+               quadrature.weight(q);
```

Comment:
Yes, I think this is really the only necessary change, at least for primitive vector-valued elements